ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-commands.sh

```diff
@@ -0,0 +1,132 @@
+    cd /tmp && tar -xvf "/tmp/${HYPERSHIFT_NAME}.tar.gz"
+    chmod +x "/tmp/${HYPERSHIFT_NAME}"
+    HCP_CLI="/tmp/${HYPERSHIFT_NAME}"
+    cd -
+  fi
+fi
+
+
+if [[ ! -f $HCP_CLI ]]; then
+  # we have to fall back to hypershift in cases where the new hcp cli isn't available yet
+  HCP_CLI="/usr/bin/hypershift"
+fi
+echo "Using $HCP_CLI for cli"
+
+RUN_HOSTEDCLUSTER_CREATION="${RUN_EXTERNAL_INFRA_TEST:-$RUN_HOSTEDCLUSTER_CREATION}"
+
+if [ "${RUN_HOSTEDCLUSTER_CREATION}" != "true" ]
+then
+  echo "Creation of a kubevirt hosted cluster has been skipped."
+  exit 0
+fi
+
+EXTRA_ARGS=""
+
+if [ -n "${KUBEVIRT_CSI_INFRA}" ]
+then
+  EXTRA_ARGS="${EXTRA_ARGS} --infra-storage-class-mapping=${KUBEVIRT_CSI_INFRA}/${KUBEVIRT_CSI_INFRA}"
+fi
+
+if [ "$(oc get infrastructure cluster -o=jsonpath='{.status.platformStatus.type}')" == "AWS" ]; then
+  if [ -z "$ETCD_STORAGE_CLASS" ]; then
+    echo "AWS infra detected. Setting --etcd-storage-class"
+    ETCD_STORAGE_CLASS="gp3-csi"
+  fi
+fi
+
+if [ -n "${ETCD_STORAGE_CLASS}" ]
+then
+  EXTRA_ARGS="${EXTRA_ARGS} --etcd-storage-class=${ETCD_STORAGE_CLASS}"
+fi
+
+PULL_SECRET_PATH="/etc/ci-pull-credentials/.dockerconfigjson"
+ICSP_COMMAND=""
+if [[ $ENABLE_ICSP == "true" ]]; then
+  ICSP_COMMAND=$(echo "--image-content-sources ${SHARED_DIR}/mgmt_icsp.yaml")
+  echo "extract secret/pull-secret"
+  oc extract secret/pull-secret -n openshift-config --to=/tmp --confirm
+  PULL_SECRET_PATH="/tmp/.dockerconfigjson"
+fi
+
+# Enable wildcard routes on the management cluster
+oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p \
+  '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'
+
+
+echo "$(date) Creating HyperShift guest cluster ${CLUSTER_NAME}"
+# shellcheck disable=SC2086
+"${HCP_CLI}" create cluster kubevirt ${EXTRA_ARGS} ${ICSP_COMMAND} \
+  --name "${CLUSTER_NAME}" \
+  --namespace "${CLUSTER_NAMESPACE_PREFIX}" \
+  --node-pool-replicas "${HYPERSHIFT_NODE_COUNT}" \
+  --memory "${HYPERSHIFT_NODE_MEMORY}Gi" \
+  --cores "${HYPERSHIFT_NODE_CPU_CORES}" \
+  --root-volume-size 64 \
+  --release-image "${RELEASE_IMAGE_LATEST}" \
+  --pull-secret "${PULL_SECRET_PATH}" \
+  --generate-ssh
+
+if [[ -n ${MCE} ]] ; then
+  oc annotate hostedclusters -n "${CLUSTER_NAMESPACE_PREFIX}" "${CLUSTER_NAME}" "cluster.open-cluster-management.io/managedcluster-name=${CLUSTER_NAME}" --overwrite
+  oc apply -f - <<EOF
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  annotations:
+    import.open-cluster-management.io/hosting-cluster-name: local-cluster
+    import.open-cluster-management.io/klusterlet-deploy-mode: Hosted
+    open-cluster-management/created-via: other
+  labels:
+    cloud: auto-detect
+    cluster.open-cluster-management.io/clusterset: default
+    name: ${CLUSTER_NAME}
+    vendor: OpenShift
+  name: ${CLUSTER_NAME}
+spec:
+  hubAcceptsClient: true
+  leaseDurationSeconds: 60
+EOF
+fi
```

Comment:
Sorry, I forgot to update this portion of the code. Kubevirt should also align with AWS and the agent: https://github.com/openshift/release/blob/3c350c6d55a09d75438174c03b49cb582cb01ec1/ci-operator/step-registry/hypershift/mce/aws/create/hypershift-mce-aws-create-chain.yaml#L64-L85

Reply:
done